src/components/carbon-provider/top-modal-context.tsx

```diff
@@ -26,19 +29,19 @@ export const TopModalContextProvider = ({
   const isFirstRender = useRef(true);
 
   if (isFirstRender.current) {
-    if (!window.__CARBON_INTERNALS_MODAL_SETTER_LIST) {
-      window.__CARBON_INTERNALS_MODAL_SETTER_LIST = [];
+    if (!windowAsAny.__CARBON_INTERNALS_MODAL_SETTER_LIST) {
+      windowAsAny.__CARBON_INTERNALS_MODAL_SETTER_LIST = [];
     }
 
-    window.__CARBON_INTERNALS_MODAL_SETTER_LIST.push(setTopModal);
+    windowAsAny.__CARBON_INTERNALS_MODAL_SETTER_LIST.push(setTopModal);
 
     isFirstRender.current = false;
   }
 
   useEffect(() => {
     return () => {
-      window.__CARBON_INTERNALS_MODAL_SETTER_LIST = window.__CARBON_INTERNALS_MODAL_SETTER_LIST?.filter(
-        (setter) => setter !== setTopModal
+      windowAsAny.__CARBON_INTERNALS_MODAL_SETTER_LIST = windowAsAny.__CARBON_INTERNALS_MODAL_SETTER_LIST?.filter(
+        (setter: any) => setter !== setTopModal
```

Comment:
```suggestion
        (setter: React.Dispatch<React.SetStateAction<HTMLElement | null>>) => setter !== setTopModal
```
This is the inferred type that IntelliSense comes out with. Will this work in place of `any`?